### DISCUSSION TITLE
Broadcasting to guests

### DISCUSSION THREAD

**Discussion by aleppik** [epoptes/epoptes#112] [Q&A]
I have set up my computer class according to this wiki guide
https://github.com/ltsp/community/wiki/Single-guest-session-over-NFS

Everything is working fine except Epoptes broadcasting to single guest session (autologin) clients.
The clients section in ltsp.conf is exactly the same as in wiki guide, but applied only to classroom computers (identified by mac addresses). I can broadcast from server (or from other client) to other (manual login) computers.

Any advice?

**Comment by alkisg** [comment]
Hi, it would take me hours to reproduce your setup; could you come over to IRC instead so that we troubleshoot this over VNC?

Click the IRC channel link at https://epoptes.org/.

**Comment by alkisg** [comment]
Notes from IRC/VNC:

- Testing on Ubuntu MATE 19.10
- /home/guest is NFS with no-root-squash etc
- DISPLAY=:0, XAUTHORITY=/home/guest/.Xauthority

...and yet root can't access Xorg, can't run xrandr or show broadcasts etc.

Something fishy with XAUTHORITY over NFS perhaps, but it works fine here on 18.04 and 20.04, so it need more troubleshooting...

**Comment by aleppik** [comment]
Found this bug report
https://bugs.launchpad.net/ubuntu/+source/lightdm/+bug/1648107

LightDM is already able to create the xauthority file locally under $XDG_RUNTIME_DIR, using configuration option user-authority-in-system-dir. This option is off by default on Ubuntu 16.04/18.04, but can be enabled with

echo -e '[LightDM]\nuser-authority-in-system-dir=true' >/etc/lightdm/lightdm.conf.d/50-user-authority-in-system-dir.conf

Now broadcasting works. I'm not sure, how this workaround affects LTSP. Everything seems ok here.

**Comment by alkisg** [comment]
Great, thanks for sharing.

Your /home/nfs export did have the "no_root_squash" option though, so it was supposed to work, that was the strange thing.
Let's wait a couple of years to see if it happens to any other users at all, or if it was somehow specific to your setup only...

**Comment by alkisg** [comment]
Noone else reported this issue, closing.
